pkg/admissioncontroller/webhook/auth/seed/graph/eventhandler_shoot.go

```diff
@@ -105,7 +106,18 @@ func (g *graph) handleShootCreateOrUpdate(ctx context.Context, shoot *gardencore
 	}
 
 	g.addEdge(namespaceVertex, shootVertex)
-	g.addEdge(cloudProfileVertex, shootVertex)
+
+	cloudProfileReference := gardenerutils.BuildCloudProfileReference(shoot)
```

Comment:
Just noticed that https://github.com/gardener/gardener/blob/master/docs/deployment/gardenlet_api_access.md should be updated, too 😃

Reply:
Also have that on the radar, will do so in the followup PR when enabling the `NamespacedCloudProfiles` (outlook: see  https://github.com/LucaBernstein/gardener/commit/c797b81d51c34ea069c57bef686fd89d0e700535#diff-38b3b798b6ee96d51b7979c0ec1dbbd570008f8f5ed1902e0ac469ceef90c3dcR63  😉 )